packages/filesystem/src/browser/file-tree/file-tree-model.ts

```diff
@@ -148,7 +148,25 @@ export class FileTreeModel extends TreeModel implements LocationService {
         if (!await this.fileSystem.exists(uri)) {
             await this.fileSystem.createFolder(uri);
         }
-        reader.readEntries(items => this.uploadEntries(newBase, items as any));
+        this.readEntries(reader, newBase);
+    }
+
+    protected readEntries(reader: WebKitDirectoryReader, dest: URI): void {
+        let allFiles = [] as WebKitFileEntry[];
```

Comment:
Do you need it? Will it be possible to upload in the reader callback?

Reply:
Added a comment to the readEntries method,
Put the upload within the callback.